### PR TITLE
Restructured vector dof reflections and prevent form compilation warnings

### LINKED ISSUE #207: warnings on form compilation
I am getting several warnings when compiling forms from dolfinx.

Below is the output when running `dolfinx/python/demo/stokes-taylor-hood/demo_stokes-taylor-hood.py`

```
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c: In function ‘create_dofmap_5f40f9c775897ede8b4bf82993c85484363aafb7’:
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c:2534:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c: In function ‘create_dofmap_4c265f74c56b4aa11d9777c47c03e105bad5b0be’:
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c:2617:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c: In function ‘create_dofmap_6cb3ad097fe4be1fae82f84f3c2b45e868a2ebde’:
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c:2671:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c: In function ‘create_dofmap_cd7a6bc181fadfbdceca0887c63423446f7d34bf’:
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c:2737:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c: In function ‘tabulate_tensor_integral_cell_otherwise_28b3c200ded93f24df4749a4d01805a62878ea3d’:
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c:2949:15: warning: unused variable ‘FE8_C0_D10_Q3_dofmap’ [-Wunused-variable]
     const int FE8_C0_D10_Q3_dofmap[5] = { 6, 7, 9, 10, 11 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c:2948:15: warning: unused variable ‘FE13_C0_D01_Q3_dofmap’ [-Wunused-variable]
     const int FE13_C0_D01_Q3_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~~
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c:2947:15: warning: unused variable ‘FE8_C0_D01_Q3_dofmap’ [-Wunused-variable]
     const int FE8_C0_D01_Q3_dofmap[5] = { 6, 8, 9, 10, 11 };
               ^~~~~~~~~~~~~~~~~~~~
At top level:
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c:2947:15: warning: ‘FE8_C0_D01_Q3_dofmap’ defined but not used [-Wunused-const-variable=]
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c:2948:15: warning: ‘FE13_C0_D01_Q3_dofmap’ defined but not used [-Wunused-const-variable=]
     const int FE13_C0_D01_Q3_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~~
libffcx_forms_11736d4fcd6f4c7d7fda49a599db40d5a6f106f4.c:2949:15: warning: ‘FE8_C0_D10_Q3_dofmap’ defined but not used [-Wunused-const-variable=]
     const int FE8_C0_D10_Q3_dofmap[5] = { 6, 7, 9, 10, 11 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c: In function ‘create_dofmap_5f40f9c775897ede8b4bf82993c85484363aafb7’:
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c:2534:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c: In function ‘create_dofmap_4c265f74c56b4aa11d9777c47c03e105bad5b0be’:
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c:2617:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c: In function ‘create_dofmap_6cb3ad097fe4be1fae82f84f3c2b45e868a2ebde’:
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c:2671:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c: In function ‘create_dofmap_cd7a6bc181fadfbdceca0887c63423446f7d34bf’:
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c:2737:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c: In function ‘tabulate_tensor_integral_cell_otherwise_88dcad742c4987f5db3d534023dc372e8d003462’:
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c:2949:15: warning: unused variable ‘FE4_C0_D10_Q3_dofmap’ [-Wunused-variable]
     const int FE4_C0_D10_Q3_dofmap[5] = { 0, 1, 3, 4, 5 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c:2948:15: warning: unused variable ‘FE9_C0_D01_Q3_dofmap’ [-Wunused-variable]
     const int FE9_C0_D01_Q3_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c:2947:15: warning: unused variable ‘FE4_C0_D01_Q3_dofmap’ [-Wunused-variable]
     const int FE4_C0_D01_Q3_dofmap[5] = { 0, 2, 3, 4, 5 };
               ^~~~~~~~~~~~~~~~~~~~
At top level:
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c:2947:15: warning: ‘FE4_C0_D01_Q3_dofmap’ defined but not used [-Wunused-const-variable=]
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c:2948:15: warning: ‘FE9_C0_D01_Q3_dofmap’ defined but not used [-Wunused-const-variable=]
     const int FE9_C0_D01_Q3_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_8c792bd8ee5ca4662907615f375231a3830f9f86.c:2949:15: warning: ‘FE4_C0_D10_Q3_dofmap’ defined but not used [-Wunused-const-variable=]
     const int FE4_C0_D10_Q3_dofmap[5] = { 0, 1, 3, 4, 5 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c: In function ‘create_dofmap_5f40f9c775897ede8b4bf82993c85484363aafb7’:
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c:2534:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c: In function ‘create_dofmap_4c265f74c56b4aa11d9777c47c03e105bad5b0be’:
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c:2617:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c: In function ‘create_dofmap_6cb3ad097fe4be1fae82f84f3c2b45e868a2ebde’:
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c:2671:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c: In function ‘create_dofmap_cd7a6bc181fadfbdceca0887c63423446f7d34bf’:
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c:2737:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c: In function ‘tabulate_tensor_integral_cell_otherwise_05ebde50c08aa524543ae80d6f60f715d05f8ebc’:
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c:2949:15: warning: unused variable ‘FE4_C0_D10_Q3_dofmap’ [-Wunused-variable]
     const int FE4_C0_D10_Q3_dofmap[5] = { 0, 1, 3, 4, 5 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c:2948:15: warning: unused variable ‘FE9_C0_D01_Q3_dofmap’ [-Wunused-variable]
     const int FE9_C0_D01_Q3_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c:2947:15: warning: unused variable ‘FE4_C0_D01_Q3_dofmap’ [-Wunused-variable]
     const int FE4_C0_D01_Q3_dofmap[5] = { 0, 2, 3, 4, 5 };
               ^~~~~~~~~~~~~~~~~~~~
At top level:
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c:2947:15: warning: ‘FE4_C0_D01_Q3_dofmap’ defined but not used [-Wunused-const-variable=]
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c:2948:15: warning: ‘FE9_C0_D01_Q3_dofmap’ defined but not used [-Wunused-const-variable=]
     const int FE9_C0_D01_Q3_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_6918ba168062b5b12715984fb74bfe8a4db77dbf.c:2949:15: warning: ‘FE4_C0_D10_Q3_dofmap’ defined but not used [-Wunused-const-variable=]
     const int FE4_C0_D10_Q3_dofmap[5] = { 0, 1, 3, 4, 5 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_7fca636145db150089bd872c160eea3e79d0539d.c: In function ‘create_dofmap_6cb3ad097fe4be1fae82f84f3c2b45e868a2ebde’:
libffcx_forms_7fca636145db150089bd872c160eea3e79d0539d.c:1353:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_7fca636145db150089bd872c160eea3e79d0539d.c: In function ‘create_dofmap_cd7a6bc181fadfbdceca0887c63423446f7d34bf’:
libffcx_forms_7fca636145db150089bd872c160eea3e79d0539d.c:1419:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_7fca636145db150089bd872c160eea3e79d0539d.c: In function ‘tabulate_tensor_integral_cell_otherwise_a89ffd7fa64fbef2b1e48ccbb5922f5d8ad127e1’:
libffcx_forms_7fca636145db150089bd872c160eea3e79d0539d.c:1629:15: warning: unused variable ‘FE4_C0_D01_Q3_dofmap’ [-Wunused-variable]
     const int FE4_C0_D01_Q3_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~
At top level:
libffcx_forms_7fca636145db150089bd872c160eea3e79d0539d.c:1629:15: warning: ‘FE4_C0_D01_Q3_dofmap’ defined but not used [-Wunused-const-variable=]
libffcx_forms_aa4b3c2db1e22338f29792b4f20e37d454e5a833.c: In function ‘create_dofmap_5f40f9c775897ede8b4bf82993c85484363aafb7’:
libffcx_forms_aa4b3c2db1e22338f29792b4f20e37d454e5a833.c:2534:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_aa4b3c2db1e22338f29792b4f20e37d454e5a833.c: In function ‘create_dofmap_4c265f74c56b4aa11d9777c47c03e105bad5b0be’:
libffcx_forms_aa4b3c2db1e22338f29792b4f20e37d454e5a833.c:2617:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_aa4b3c2db1e22338f29792b4f20e37d454e5a833.c: In function ‘create_dofmap_6cb3ad097fe4be1fae82f84f3c2b45e868a2ebde’:
libffcx_forms_aa4b3c2db1e22338f29792b4f20e37d454e5a833.c:2671:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_aa4b3c2db1e22338f29792b4f20e37d454e5a833.c: In function ‘create_dofmap_cd7a6bc181fadfbdceca0887c63423446f7d34bf’:
libffcx_forms_aa4b3c2db1e22338f29792b4f20e37d454e5a833.c:2737:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_aa4b3c2db1e22338f29792b4f20e37d454e5a833.c: In function ‘tabulate_tensor_integral_cell_otherwise_46a0022a2a4a29bed333e75533aa9ce1066223a1’:
libffcx_forms_aa4b3c2db1e22338f29792b4f20e37d454e5a833.c:2948:15: warning: unused variable ‘FE7_C0_D01_Q3_dofmap’ [-Wunused-variable]
     const int FE7_C0_D01_Q3_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_aa4b3c2db1e22338f29792b4f20e37d454e5a833.c:2947:15: warning: unused variable ‘FE2_C0_Q3_dofmap’ [-Wunused-variable]
     const int FE2_C0_Q3_dofmap[6] = { 6, 7, 8, 9, 10, 11 };
               ^~~~~~~~~~~~~~~~
At top level:
libffcx_forms_aa4b3c2db1e22338f29792b4f20e37d454e5a833.c:2947:15: warning: ‘FE2_C0_Q3_dofmap’ defined but not used [-Wunused-const-variable=]
libffcx_forms_aa4b3c2db1e22338f29792b4f20e37d454e5a833.c:2948:15: warning: ‘FE7_C0_D01_Q3_dofmap’ defined but not used [-Wunused-const-variable=]
     const int FE7_C0_D01_Q3_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_dbac2d0e737e185e2e64a7564b2fa1015ee9ac58.c: In function ‘create_dofmap_6cb3ad097fe4be1fae82f84f3c2b45e868a2ebde’:
libffcx_forms_dbac2d0e737e185e2e64a7564b2fa1015ee9ac58.c:1353:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_dbac2d0e737e185e2e64a7564b2fa1015ee9ac58.c: In function ‘create_dofmap_cd7a6bc181fadfbdceca0887c63423446f7d34bf’:
libffcx_forms_dbac2d0e737e185e2e64a7564b2fa1015ee9ac58.c:1419:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_dbac2d0e737e185e2e64a7564b2fa1015ee9ac58.c: In function ‘tabulate_tensor_integral_cell_otherwise_8c34aa9664a7a7322c85b0c3937d1ac6ee8212e1’:
libffcx_forms_dbac2d0e737e185e2e64a7564b2fa1015ee9ac58.c:1629:15: warning: unused variable ‘FE4_C0_D01_Q1_dofmap’ [-Wunused-variable]
     const int FE4_C0_D01_Q1_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~
At top level:
libffcx_forms_dbac2d0e737e185e2e64a7564b2fa1015ee9ac58.c:1629:15: warning: ‘FE4_C0_D01_Q1_dofmap’ defined but not used [-Wunused-const-variable=]
(A) Norm of velocity coefficient vector: 17.435407162111872
(A) Norm of pressure coefficient vector: 311.4425381292949
(B) Norm of velocity coefficient vector: 17.435407193004238
(B) Norm of pressure coefficient vector: 311.4425379790811
(C) Norm of velocity coefficient vector: 17.435407174711123
(C) Norm of pressure coefficient vector: 311.4425379789682
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c: In function ‘create_dofmap_6cb3ad097fe4be1fae82f84f3c2b45e868a2ebde’:
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c:3369:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c: In function ‘create_dofmap_cd7a6bc181fadfbdceca0887c63423446f7d34bf’:
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c:3435:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c: In function ‘create_dofmap_5f40f9c775897ede8b4bf82993c85484363aafb7’:
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c:3503:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c: In function ‘create_dofmap_4c265f74c56b4aa11d9777c47c03e105bad5b0be’:
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c:3586:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c: In function ‘create_dofmap_7f791dfd60e9f19bc1db3b8814f5fb665066603d’:
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c:3673:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c: In function ‘tabulate_tensor_integral_cell_otherwise_8cd58b3d9efced44aabfaf43246e21efbd3b6de9’:
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c:3886:15: warning: unused variable ‘FE16_C2_Q6_dofmap’ [-Wunused-variable]
     const int FE16_C2_Q6_dofmap[3] = { 12, 13, 14 };
               ^~~~~~~~~~~~~~~~~
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c:3885:15: warning: unused variable ‘FE16_C0_D10_Q6_dofmap’ [-Wunused-variable]
     const int FE16_C0_D10_Q6_dofmap[5] = { 6, 7, 9, 10, 11 };
               ^~~~~~~~~~~~~~~~~~~~~
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c:3884:15: warning: unused variable ‘FE4_C0_D01_Q6_dofmap’ [-Wunused-variable]
     const int FE4_C0_D01_Q6_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c:3883:15: warning: unused variable ‘FE16_C0_D01_Q6_dofmap’ [-Wunused-variable]
     const int FE16_C0_D01_Q6_dofmap[5] = { 6, 8, 9, 10, 11 };
               ^~~~~~~~~~~~~~~~~~~~~
At top level:
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c:3883:15: warning: ‘FE16_C0_D01_Q6_dofmap’ defined but not used [-Wunused-const-variable=]
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c:3884:15: warning: ‘FE4_C0_D01_Q6_dofmap’ defined but not used [-Wunused-const-variable=]
     const int FE4_C0_D01_Q6_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c:3885:15: warning: ‘FE16_C0_D10_Q6_dofmap’ defined but not used [-Wunused-const-variable=]
     const int FE16_C0_D10_Q6_dofmap[5] = { 6, 7, 9, 10, 11 };
               ^~~~~~~~~~~~~~~~~~~~~
libffcx_forms_acc14651a3453b4668543df8072663bceec24af5.c:3886:15: warning: ‘FE16_C2_Q6_dofmap’ defined but not used [-Wunused-const-variable=]
     const int FE16_C2_Q6_dofmap[3] = { 12, 13, 14 };
               ^~~~~~~~~~~~~~~~~
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c: In function ‘create_dofmap_6cb3ad097fe4be1fae82f84f3c2b45e868a2ebde’:
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c:3369:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c: In function ‘create_dofmap_cd7a6bc181fadfbdceca0887c63423446f7d34bf’:
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c:3435:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c: In function ‘create_dofmap_5f40f9c775897ede8b4bf82993c85484363aafb7’:
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c:3503:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c: In function ‘create_dofmap_4c265f74c56b4aa11d9777c47c03e105bad5b0be’:
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c:3586:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c: In function ‘create_dofmap_7f791dfd60e9f19bc1db3b8814f5fb665066603d’:
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c:3673:29: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   dofmap->base_permutations = bp;
                             ^
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c: In function ‘tabulate_tensor_integral_cell_otherwise_57c82f8664b849190604db35b2df5d301f5ebff1’:
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c:3884:15: warning: unused variable ‘FE4_C0_D01_Q6_dofmap’ [-Wunused-variable]
     const int FE4_C0_D01_Q6_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c:3883:15: warning: unused variable ‘FE7_C0_Q6_dofmap’ [-Wunused-variable]
     const int FE7_C0_Q6_dofmap[6] = { 6, 7, 8, 9, 10, 11 };
               ^~~~~~~~~~~~~~~~
At top level:
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c:3883:15: warning: ‘FE7_C0_Q6_dofmap’ defined but not used [-Wunused-const-variable=]
libffcx_forms_408d57d0c1038d2e170a2b3e84c13090edccb92b.c:3884:15: warning: ‘FE4_C0_D01_Q6_dofmap’ defined but not used [-Wunused-const-variable=]
     const int FE4_C0_D01_Q6_dofmap[2] = { 3, 4 };
               ^~~~~~~~~~~~~~~~~~~~
(D) Norm of velocity coefficient vector: 17.43540717470972
(D) Norm of pressure coefficient vector: 311.59141819987246
```

The one for `dofmap->base_permutations = bp;` was probably introduced by #206.
I am not able to pinpoint the exact cause for the other two, but I can assure that none of them were happening three days ago, so it's possible that this is due to some of the modifications introduced in either #201 or #204.
Thanks

### PR DESCRIPTION
**Pull request by mscroggs**
- Fixes #207.
- Remove global variables in `codegeneration/utils.py`:
https://github.com/FEniCS/ffcx/blob/c2e4b518985d52124499e734de93a79d025bbadd/ffcx/codegeneration/utils.py#L70-L71
- Moved vector reflection and table dofmap functions out of `codegeneration/utils.py`, and slightly simplified them in the two places they are used.

### CODE REVIEW

File: ffcx/codegeneration/integrals_generator.py [hunk 1]
```diff
@@ -149,12 +148,49 @@ def generate(self):
                           "coordinate_dofs = (const double*)__builtin_assume_aligned(coordinate_dofs, {});"
                           .format(alignment))]
 
+        self.contains_reflections = {}
+        c_false = L.LiteralBool(False)
         for element, id in self.ir.element_ids.items():
-            parts += get_vector_reflection_array(L, self.ir.element_dof_reflection_entities[element],
-                                                 "ref_dof" + str(id))
-
+            vname = "ref_dof" + str(id)
```
Comment (michalhabera) [inline ffcx/codegeneration/integrals_generator.py]: Make this part which generates a reflections array a member function of IntegralGenerator. The reason is to have the same granularity of code in generate() methods - if you have a look around this code there are only codes to other generate routines.

File: ffcx/codegeneration/integrals_generator.py [hunk 2]
```diff
@@ -149,12 +148,49 @@ def generate(self):
...
+                        # Mark this space as needing reflections
+                        self.contains_reflections[id] = True
+
+            # If no dofs need reflecting, don't write any array
+            if self.contains_reflections[id]:
+                parts.append(L.ArrayDecl(
+                    "const bool", L.Symbol(vname), (len(reflect_dofs), ), values=reflect_dofs))
+
+        self.table_dofmaps = {}
         for tname, dofmap in self.ir.table_dofmaps.items():
```
Comment (michalhabera) [inline ffcx/codegeneration/integrals_generator.py]: Similar here, make `IntegralGenerator.generate_table_dofmaps` method to keep the same code granularity.

File: ffcx/codegeneration/symbols.py [hunk 3]
```diff
@@ -92,6 +92,17 @@ def argument_loop_index(self, iarg):
         indices = ["i", "j", "k", "l"]
         return self.S(indices[iarg])
 
+    def get_reflection(self, L, i):
```
Comment (michalhabera) [inline ffcx/codegeneration/symbols.py]: Just `reflection` here, other FFCXBackendSymbols doesnt have `get_*` format.

Comment (mscroggs) [reply ffcx/codegeneration/symbols.py]: I've renamed it `entity_reflection`, as `reflection` could mean a lot of things

**Pull request merged**